sumdbaudit/README.md

```diff
@@ -119,6 +122,38 @@ In the `ExecStart` line above, add `-unpack` and then restart the `sumdbmirror`
 When it next updates tiles this table will be populated.
 This will use more CPU and around 60% more disk.
 
+## Setting up a `witness` service
+This requires a local clone of the SumDB Log to be available. For this to be of any
+real value, it should be running against a database which is regularly being updated
+by the `mirror` service described above.
+
+A client which successfully checks its checkpoints with a witness can ensure that if
+there is a "split view" of the SumDB Log, then it is on the same side of the split as
+the witness. If this witness is also verifying the claims of the log, then the client
+is safe in relying on the data within (providing it trusts the verifer!).
+
+The service can be started with the command (assuming `~/sum.db` is the database):
+```bash
+go run ./sumdbaudit/cli/witness -listen :8080 -db ~/sum.db -v=1 -alsologtostderr
+```
+
+This can be set up as a Linux service in much the same way as the `mirror` above.
+
+Once running, the server will be available for GET requests at the listen address
+given as a commandline parameter.
+
+Some example requests that can be made:
+```bash
+# Simply get the latest golden checkpoint
+curl -i http://localhost:8080/golden
+
+# Validate that the witness is consistent with the Checkpoint your go build tools are using
```

Comment:
It would be better to fetch the latest "consistent" STH from the witness, and have the local tooling verify consistency as the witness could be dishonest about the result...
Perhaps the witness could return its latest golden+proof to the supplied STH and then the client can reverify?

Reply:
I've added notes in the README and TODOs in the code to add such proofs. The README now makes it clear that this witness is only suitable for use where it is trusted and not a general purpose adversarial situation.

This would be great to add but would be a significant amount of work right now. It needs:
 * The Missing Feature to be added where we download partial tiles (OR, accept that this will make an outbound request to the log for every inbound request)
 * An API defining
 * A verifying client
 * Code to synthesize the consistency proof

This is all doable, but I'm going to heavily lean on pragmatism and say that this code still provides some value that is otherwise impossible to realize in the current ecosystem (there is currently no way for 2 parties to check that they are seeing consistent views of the log unless they get the same checkpoint), and thus let's launch and iterate.

Reply:
SGTM - my only fear was someone "cut'n'pasting" this without understanding the context, and ending shooting themselves in the foot. If we're calling it out in the README as coming with some caveats then that's cool.